org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/ScalaIndenter.scala

```diff
@@ -1518,10 +1517,9 @@ class ScalaIndenter(
    */
   private def isStringOrCharLiteralAssignment(referenceTokenPos: Int, offset: Int) = {
     val restOfTheLine = document.get(referenceTokenPos, Math.max(offset - referenceTokenPos, 0))
-    val charLit = """'(.|\\\w|\\\\)'"""
-    val stringLit = """"(\\"|[^"])*")"""
-
-    val regex = s"(${charLit}|${stringLit}$$".r
+    val charLit = "'.'"
+    val stringLit = "\".*\""
```

Comment:
The only important rule is that we do an indent when the caret is after the `=` but before the rhs. The content of the rhs shouldn't matter.